client/src/test/scala/org/http4s/client/middleware/FollowRedirectSpec.scala

```diff
@@ -2,39 +2,47 @@ package org.http4s
 package client
 package middleware
 
-import org.http4s.Status._
-import org.http4s.Method._
-import org.http4s.headers.Location
+import java.util.concurrent.atomic._
 
+import org.http4s.dsl._
+import org.http4s.headers._
+import scalaz.concurrent._
 
 class FollowRedirectSpec extends Http4sSpec {
 
-  val route = HttpService {
-    case r if r.method == GET && r.pathInfo == "/ok"       => Response(Ok).withBody("hello")
-    case r if r.method == GET && r.pathInfo == "/redirect" => Response(MovedPermanently).replaceAllHeaders(Location(uri("/ok"))).withBody("Go there.")
-    case r if r.method == GET && r.pathInfo == "/loop"     => Response(MovedPermanently).replaceAllHeaders(Location(uri("/loop"))).withBody("Go there.")
-    case r if r.method == POST && r.pathInfo == "/303"      => 
-      Response(SeeOther).replaceAllHeaders(Location(uri("/ok"))).withBody("Go to /ok")
+  private val loopCounter = new AtomicInteger(0)
 
-    case r => sys.error("Path not found: " + r.pathInfo)
+  val service = HttpService {
+    case GET -> Root / "ok" =>
+      Response(Ok).withBody("hello")
+    case GET -> Root / "redirect" =>
+      Response(MovedPermanently).replaceAllHeaders(Location(uri("/ok"))).withBody("Go there.")
+    case GET -> Root / "loop" =>
+      Response(MovedPermanently).replaceAllHeaders(Location(uri("/loop"))).withBody(loopCounter.incrementAndGet.toString)
```

Comment:
Since "/loop" is now stateful (the AtomicInteger) could we move that test route and its atomic into the test so that nobody is tempted to re-run it and find that the counter has started at 4?